ingest/perf_test_ingestor.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 Rackspace US, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ingest
+
+import (
+	"context"
+	"fmt"
+	"github.com/racker/salus-telemetry-protocol/telemetry_edge"
+	"github.com/racker/telemetry-envoy/ambassador"
+	"github.com/racker/telemetry-envoy/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type PerfTestIngestor struct {
+	egressConn       ambassador.EgressConnection
+	metricsPerMinute int
+	floatsPerMetric  int
+	ticker           *time.Ticker
+	newRateC         chan int
+}
+
+func init() {
+	registerIngestor(&PerfTestIngestor{})
+}
+
+func (p *PerfTestIngestor) Bind(conn ambassador.EgressConnection) error {
+	if viper.GetInt(config.PerfTestPort) == 0 {
+		return nil
+	}
+	log.Info("entering perfTest mode")
+	p.egressConn = conn
+	p.metricsPerMinute = 60
+	p.floatsPerMetric = 10
+	p.newRateC = make(chan int)
+	return nil
+}
+
+func (p *PerfTestIngestor) Start(ctx context.Context) {
+	if viper.GetInt(config.PerfTestPort) == 0 {
+		return
+	}
+
+	p.ticker = time.NewTicker(time.Duration(int64(time.Minute) / int64(p.metricsPerMinute)))
+	go p.startPerfTestServer()
+	for {
+		select {
+		case <-ctx.Done():
+			p.ticker.Stop()
+			return
+		case p.metricsPerMinute = <-p.newRateC:
+			p.ticker.Stop()
+			p.ticker = time.NewTicker(time.Duration(int64(time.Minute) / int64(p.metricsPerMinute)))
+		case <-p.ticker.C:
+			p.processMetric()
+		}
+	}
+}
+
+func (p *PerfTestIngestor) startPerfTestServer() {
+	serverMux := http.NewServeMux()
+	serverMux.Handle("/", http.HandlerFunc(p.handler))
```

Comment:
If you want to shave off a few characters, this seems to work
```suggestion
	serverMux.HandleFunc("/", p.handler)
```

however, I'm now confused why they also have `http.HandlerFunc`. I guess it's for middleware that accepts only `http.Handler` instances.